editor/components/EditCrashAddressModal.tsx

```diff
@@ -0,0 +1,255 @@
+      }
+    );
+  }, [crash]);
+
+  const {
+    register,
+    handleSubmit,
+    reset,
+    getValues,
+    setValue,
+    formState: { isDirty },
+  } = useForm<AddressFormInputs>({
+    defaultValues: defaultValues,
+  });
+
+  const { mutate } = useMutation(UPDATE_CRASH);
+
+  const primaryColumns = [
+    crashesColumns.rpt_block_num,
+    crashesColumns.rpt_street_pfx,
+    crashesColumns.rpt_street_name,
+    crashesColumns.rpt_street_sfx,
+    crashesColumns.rpt_street_desc,
+    crashesColumns.rpt_road_part_id,
+    crashesColumns.rpt_rdwy_sys_id,
+    crashesColumns.rpt_hwy_num,
+  ];
+  const secondaryColumns = [
+    crashesColumns.rpt_sec_block_num,
+    crashesColumns.rpt_sec_street_pfx,
+    crashesColumns.rpt_sec_street_name,
+    crashesColumns.rpt_sec_street_sfx,
+    crashesColumns.rpt_sec_street_desc,
+    crashesColumns.rpt_sec_road_part_id,
+    crashesColumns.rpt_sec_rdwy_sys_id,
+    crashesColumns.rpt_sec_hwy_num,
+  ];
+
+  /**
+   * Submits mutation to database on save button click
+   */
+  const onSave: SubmitHandler<AddressFormInputs> = async (data) => {
+    reset(data);
+    await mutate({
+      id: crashId,
+      updates: data,
+    });
+
+    onSaveCallback();
+  };
+
+  /**
+   * Component that renders the form field based on the input type
+   * of the column, such as text or select
+   */
+  const FormField = ({ col }: FormColumnProps) => {
+    const [query, typename] = useLookupQuery(
+      col?.relationship ? col.relationship : undefined
+    );
+
+    const { data: selectOptions, error: selectOptionsError } =
+      useQuery<LookupTableOption>({
+        query,
+        // we don't need to refetch lookup table options
+        options: { revalidateIfStale: false },
+        typename,
+      });
+    const inputType = col.inputType;
+    return (
+      <div className="mb-2">
+        {(inputType === "text" || inputType === "number") && (
+          <Form.Control
+            {...register(col.path)}
+            size="sm"
+            type="text"
+            inputMode={inputType === "number" ? "numeric" : undefined}
+          />
+        )}
+        {inputType === "select" && (selectOptions || !!selectOptionsError) && (
+          <Form.Select
+            {...register(col.path)}
+            size="sm"
+            disabled={!selectOptions}
+          >
+            <option value="">Select...</option>
+            {selectOptions?.map((option) => (
+              <option key={option.id} value={String(option.id)}>
+                {option.label}
+              </option>
+            ))}
+          </Form.Select>
+        )}
+      </div>
+    );
+  };
+
+  return (
+    <Modal
+      show={show}
+      onHide={() => {
+        reset();
+        setShowEditAddressModal(false);
+      }}
+      size="lg"
+    >
+      <Modal.Header>
+        <Modal.Title className="me-2">Edit crash address</Modal.Title>
+        <CrashSwapAddressButton getValues={getValues} setValue={setValue} />
```

Comment:
Would you make this button aligned to the right side of the modal header? It now looks off to me on the left.